src/AutoPilotPlugins/APM/APMSensorsComponent.qml

```diff
@@ -446,6 +446,34 @@ SetupPage {
                                 model:      _orientationsDialogShowCompass ? 3 : 0
                                 delegate:   singleCompassSettingsComponent
                             }
+
+                            QGCLabel {
+                                width:      parent.width
+                                visible:    globals.activeVehicle.sub && _orientationsDialogShowCompass
+                                wrapMode:   Text.WordWrap
+                                text:       qsTr("Magnetic Declination")
+                            }
+
+                            Column {
+                                id:                 declinationColumnLayout
+                                visible:            globals.activeVehicle.sub && _orientationsDialogShowCompass
+                                anchors.margins:    ScreenTools.defaultFontPixelWidth
+                                anchors.left:       parent.left
+                                anchors.right:      parent.right
+                                spacing:            ScreenTools.defaultFontPixelHeight
+
+                                FactCheckBox {
+                                    id:         manualMagneticDeclination
+                                    text:       qsTr("Automatic Declination")
+                                    fact:       sensorParams.autoDeclinationFact
+                                    visible:    true
+                                }
+
+                                FactTextField {
+                                    fact:       sensorParams.declinationFact
+                                    visible:    !sensorParams.autoDeclinationFact.value
```

Comment:
The pattern for things like this more follows disabling if the controlling checkbox is in a state. Making things come/go visually can be more confusing to folks.

Reply:
ok, fixed